config/profile.go

```diff
@@ -651,15 +652,19 @@ func (p *Profile) SetRootPath(rootPath string) {
 func (p *Profile) resolveSourcePath(sourceBase string, sourcePaths ...string) []string {
 	var applySourceBase, applyBaseDir pathFix
 
-	// Backup source is NOT relative to the configuration, but to PWD or sourceBase (if not empty)
-	// Applying "sourceBase" if set
-	if sourceBase = strings.TrimSpace(sourceBase); sourceBase != "" {
-		sourceBase = fixPath(sourceBase, expandEnv, expandUserHome)
-		applySourceBase = absolutePrefix(sourceBase)
-	}
-	// Applying a custom PWD eagerly so that own commands (e.g. "show") display correct paths
-	if p.BaseDir != "" {
-		applyBaseDir = absolutePrefix(p.BaseDir)
+	sourceBase = fixPath(sourceBase, expandEnv, expandUserHome)
+	// When "source-relative" is set, the source paths are relative to the "source-base"
+	if p.Backup == nil || !p.Backup.SourceRelative {
```

Comment:
After checking it further, I can see we need an additional condition here. The method is used to resolve backup sources in the backup section and also in `SetPath`. When called in `SetPath`, it must not make the paths relative because `SetPath` sets the path args for non-backup commands and they expect absolute paths.

I suggest we change the method signature, from:

```go
func (p *Profile) resolveSourcePath(sourceBase string, sourcePaths ...string) []string {
```

to: 

```go
func (p *Profile) resolveSourcePath(sourceBase string, relativePaths bool, sourcePaths ...string) []string {
```

This also simplifies the if condition since the only caller where this is allowed to be `true` is `p.Backup`.